tests/Common/ManagerLogicTestTrait.php

```diff
@@ -854,4 +859,39 @@ public function testRevokeAll(): void
         $this->assertEmpty($this->assignmentsStorage->getByUserId('author B'));
         $this->assertSame($manager, $returnedManager);
     }
+
+    public function testDataPersistency(): void
+    {
+        $itemsStorage = new FakeItemsStorage();
+        $assignmentsStorage = new FakeAssignmentsStorage();
+        $manager = $this->createManager($itemsStorage, $assignmentsStorage);
+        $manager
+            ->addRole((new Role('role1'))->withCreatedAt(1_694_502_936)->withUpdatedAt(1_694_502_936))
+            ->addRole((new Role('role2'))->withCreatedAt(1_694_502_976)->withUpdatedAt(1_694_502_976))
+            ->addChild('role1', 'role2');
+        $manager->assign('role1', 1);
+        $manager->assign('role2', 2);
+
+        $this->assertEquals(
+            [
+                'role1' => (new Role('role1'))->withCreatedAt(1_694_502_936)->withUpdatedAt(1_694_502_936),
+                'role2' => (new Role('role2'))->withCreatedAt(1_694_502_976)->withUpdatedAt(1_694_502_976),
+            ],
+            $itemsStorage->getAll(),
+        );
+        $this->assertEquals(
+            [
+                'role1' => (new Role('role1'))->withCreatedAt(1_694_502_936)->withUpdatedAt(1_694_502_936),
+                'role2' => (new Role('role2'))->withCreatedAt(1_694_502_976)->withUpdatedAt(1_694_502_976),
+            ],
+            $manager->getChildRoles('role1'),
```

Comment:
Child roles of `role1` must be `role2` only, right?

Reply:
That's how it initially was. It needs to be fixed. I created issue earlier - #178.

Reply:
May be fix #178 before merge this PR. This suggestion for prevent create incorrect tests.

Reply:
#191